### PR TITLE
WIP: Adding the sync function folder, test files,

### PR DESCRIPTION
**Pull request by sammcdsam**
This pull request is a work in progress. 
I just wanted to get it posted so it could be reviewed for initial comments

approx_match.py, and a tester file for approx_match and tre.
tester file can successfully find matches between a given python file and html file.
I have not tested using parts of the html to search the python file. 

Question - 
Where should this code actually be? Eventually it will all be integrated, but for the time being is it ok where it is?
Should I have merged this branch into my own fork first or is it ok to create a pull request off a branch?

### CODE REVIEW

File: CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # *********************************************************************
 # |docname| - Template configuration file for a Sphinx CodeChat project
 # *********************************************************************
-# This file configures Sphinx, which transforms restructured text (reST) into
+# 40, which transforms restructured text (reST) into
```
Comment (bjones1) [inline CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py:21]: ??? Revert this change.

File: sync_func [hunk 2]
```diff
@@ -0,0 +1,450 @@
+# .. -*- coding: utf-8 -*-
```
Comment (bjones1) [inline sync_func:1]: What is this?

**Review by bjones1 — COMMENTED** [review]
You also need some info / instructions on how to build TRE, how to run these tests, etc.

**Comment by abp168** [comment]
### Instructions:
In order to run the stand-alone string matching, the user must have the following files in this pull request:

- approx_match.py

- html_example.txt

- tre_example.py

- conf.py

**Installation:**
In addition, the [tre](https://github.com/laurikari/tre/pull/49) library must be fully installed for python. The directions to install the tre library for an ubuntu system can be found [here](https://github.com/anpc/sines-in-aging/blob/78baa94af343125d1ea916f3007b652ebc2d8cac/INSTALL/libraries.sh#L12-L19). After installing the tre package, the user must call `python setup.py install` in the command line to properly setup the tre library. Make sure to connect this package to the python interpreter you plan to use if there are multiple interpreters on your system. 

**Usage:**

The conf.py will be the source code that will be rendered in code chat (when fully implemented) and the html_example.txt is the rendered html code from the conf.py source code.  

The tre_example.py file will act as a command line application that the user can run by using the following command 
`$ python tre_example.py`

The output will be 20 random instances of matching the strings in the html_example.txt to the conf.py  source code. The matching occurs by calling the approx_match module from the approx_match.py file. The output will return the rendered text and the source code were able to match and the cursor index of the match for each file.

File: CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py (hunk 1 above)
Comment (abp168) [reply CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py:21]: fixed

File: sync_func (hunk 2 above)
Comment (abp168) [reply sync_func:1]: deleted

File: CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py (hunk 1 above)
Comment (abp168) [reply CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py:21]: The fix for this and the next comment are in a new PR. Not sure if this is what you wanted ( I remember in your lecture you said to make small PRs but not sure if this is the case you wanted it for). If not, just so I know in the future, is it possible for another user to add changes to a branch they have checked out and have it show up in the same PR?

**Comment by bjones1** [comment]
Sam and Aaron, my apologies for not providing more feedback for your PR -- I'll do a better job of this for the next project. For now, this is fine.

File: CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py (hunk 1 above)
Comment (bjones1) [reply CodeChat_Server/CodeChat_Server/templates/sphinx/conf.py:21]: Sorry, missed this until now. You can always makes changes to an existing PR by pushing more changes to it.

**Comment by bjones1** [comment]
Closing in favor of #26.
